Bookish/Bookish.Web/Controllers/HomeController.cs

```diff
@@ -45,6 +45,34 @@ public IActionResult Book(string isbn)
             }
             return View(new BookViewModel(book));
         }
+        public IActionResult AddBook(string errorMessage = "")
+        {
+            return View(new AddBookViewModel(errorMessage));
+        }
+
+        [HttpPost]
+        public IActionResult AddBook(string title, string authors, string isbn, int copies)
+        {
+            var errorMessage = "";
+            if (copies < 1)
+            {
+                errorMessage += "You need to add at least one copy\n";
+            }
+
+            if (libraryService.BookExists(isbn))
+            {
+                errorMessage += "Book with that ISBN already exists";
+            }
+
+            if (errorMessage != "")
```

Comment:
You shouldn't construct the error message here. Instead just check if the input is valid and redirect to the AddBook get endpoint if it isn't 🙂